hoomd/pytest/test_simulation.py

```diff
@@ -258,3 +242,64 @@ def act(self, timestep):
         900,
         1000,
     ]
+
+
+def test_large_timestep(simulation_factory, lattice_snapshot_factory):
+    """Test that simluations suport large timestep values."""
+    sim = simulation_factory()
+    sim.timestep = 2**64 - 100
+    sim.create_state_from_snapshot(lattice_snapshot_factory())
+
+    sim.run(99)
+    assert sim.timestep == 2**64 - 1
+
+
+def test_timestep_wrap(simulation_factory, lattice_snapshot_factory):
+    """Test that time steps wrap around."""
+    sim = simulation_factory()
+    sim.timestep = 2**64 - 100
+    sim.create_state_from_snapshot(lattice_snapshot_factory())
+
+    sim.run(200)
+    assert sim.timestep == 100
+
+
+def test_initial_timestep_range(simulation_factory):
+    """Test that the initial timestep cannot be set out of range."""
+    sim = simulation_factory()
+
+    with pytest.raises(ValueError):
+        sim.timestep = -1
+
+    with pytest.raises(ValueError):
+        sim.timestep = 2**64
+
+    sim.timestep = 2**64 - 1
+
+
+def test_run_limit(simulation_factory, lattice_snapshot_factory):
+    """Test limits of Simulation.run()."""
+    sim = simulation_factory(lattice_snapshot_factory())
+
+    with pytest.raises(ValueError):
+        sim.run(2**64 - 1)
+
+    with pytest.raises(ValueError):
+        sim.run(-1)
+
+
+def test_seed(simulation_factory, lattice_snapshot_factory):
+    sim = simulation_factory()
+    assert sim.seed is None
+
+    sim.seed = 42
+    assert sim.seed == 42
+
+    sim.seed = 0x123456789abcdef
+    assert sim.seed == 0xcdef
```

Comment:
We should also test the correct warning is generated.

Reply:
I tested this manually. If you know a way to write pytest scripts that check for particular output on stderr, feel free to add it.

Reply:
Done.

Reply:
Apparently, the `capsys` fixture in pytest causes `python` to segfault in the CI environment. https://dev.azure.com/glotzerlab/hoomd-blue/_build/results?buildId=9549&view=logs&j=fdb24454-8954-52a4-1b2d-7de6a8c2d450&t=50c16307-6c3f-5086-6e7e-00e0a5068b16&l=2291

@b-butler Is it critical that we unit test for these warnings that we need to resolve the testing issue, or should I just revert the commit.

Reply:
You might be able to try `contextlib.redirect_stderr` and redirect to an internal StringIO object. https://docs.python.org/3/library/contextlib.html#contextlib.redirect_stderr

We use that in signac/signac-flow with pytest.

Reply:
I wouldn't say it is critical, I just thought it would be a nicety. I forgot this was generated in C++ and not a Python warning.

Reply:
Although it is strange that it causes a segfault.